### ISSUE TITLE
Some Motu M4 have a different usb device ID

### ISSUE THREAD

**Issue by Big-FG** [alsa-project/alsa-ucm-conf#212]
I just wanted to draw some attention regarding the Motu M4 UCM2 profie

currently USB-Audio.conf has the M4 with usb id 07fd:000b

```
if.motu-m24 {
	Condition {
		Type String
		Haystack "${CardComponents}"
		Needle "USB07fd:000b"
	}
	True.If.M4 {
		Condition {
			Type String
			Haystack "${CardLongName}"
			Needle "MOTU M4"
		}
		True.Define.ProfileName "MOTU/M4"
		False.Define.ProfileName "MOTU/M2"
	}
}
```

but mine is apparently different with 07fd:0008

`Bus 003 Device 002: ID 07fd:0008 Mark of the Unicorn M Series`

after adjusting the USB-Audio.conf with my usb ID the profile loads and works correctly. Im not sure if mine is just a weird one or how many people this may be the case for. Im fine with modifying the conf myself for my card if this is just specific to me as my unit wasnt purchased new but was a used one so Im not sure what might be different.

**Comment by puleglot** [comment]
> but mine is apparently different with 07fd:0008

This is the usb id of an old M2/M4 revisions based on XMOS controller. I have the same device.

**Comment by Big-FG** [comment]
> > but mine is apparently different with 07fd:0008
> 
> This is the usb id of an old M2/M4 revisions based on XMOS controller. I have the same device.

Ahh I see, I remember hearing it had different USB controllers (3 different ones from what ive read). I wonder of the other 2 have different IDs from each other then or its just the XMOS

**Comment by perexg** [comment]
Could you attach `alsa-info.sh --no-upload` output for the reference and check, if the USB audio device structure is similar ? Thank you.

**Comment by perexg** [comment]
A test patch: [motu-m4-0008.txt](https://github.com/alsa-project/alsa-ucm-conf/files/9527084/motu-m4-0008.txt)

```patch
diff --git a/ucm2/USB-Audio/USB-Audio.conf b/ucm2/USB-Audio/USB-Audio.conf
index fd3d947..284ac2f 100644
--- a/ucm2/USB-Audio/USB-Audio.conf
+++ b/ucm2/USB-Audio/USB-Audio.conf
@@ -71,9 +71,9 @@ If.steinberg-ur44 {
 
 If.motu-m24 {
        Condition {
-               Type String
-               Haystack "${CardComponents}"
-               Needle "USB07fd:000b"
+               Type RegexMatch
+               String "${CardComponents}"
+               Needle "USB07fd:000[8b]"
        }
        True.If.M4 {
                Condition {
```

**Comment by Big-FG** [comment]
> Could you attach `alsa-info.sh --no-upload` output for the reference and check, if the USB audio device structure is similar ? Thank you.

sure thing, the profile so far has worked flawlessly for me but just for sanity sake


[alsa-info.AOYZZ9pQhv.txt](https://github.com/alsa-project/alsa-ucm-conf/files/9528667/alsa-info.AOYZZ9pQhv.txt)

**Comment by perexg** [comment]
Thanks, the above patch was merged in ad1618cdd33f5c78ee1a0a9d3db362ffcb9f2596 .
